app/src/main/java/fr/free/nrw/commons/upload/UploadModel.java

```diff
@@ -300,7 +300,7 @@ private void markCurrentUploadVisited() {
     }
 
     String getSelectedLicense() {
-        return license;
+        return prefs.getString(Prefs.DEFAULT_LICENSE, Prefs.Licenses.CC_BY_SA_3);
```

Comment:
Haven't tested on a device, but won't this mean it always uploads with the default license? The change on line 68 will handle the default case well enough I think?